docs/Dialects/DC/RationaleDC.md

```diff
@@ -0,0 +1,43 @@
+# DC Dialect Rationale
+
+[TOC]
+
+## Introduction
+
+DC IR describes independent, unsynchronized processes communicating data through First-in First-out (FIFO) communication channels. 
+This can be implemented in many ways, such as using synchronous logic, or with processors. 
+
+The intention of DC is to model all operations required to represent such a control flow language. DC aims to be strictly a control
+flow dialect - as opposed to the Handshake dialect, which assigns control semantics to _all_ SSA values. As such, data values are
+only present in DC where they are required to model control flow.
+
+By having such a control language, the dialect aims to facilitate the construction of dataflow programs where control and data
+is explicitly separate. This enables optimization of the control and data side of the program independently, as well as mapping
+of the data side into functional units, pipelines, etc.. Furthermore, separating data and control will make it easier to reason
+about possible critical paths of either circuit, which may inform buffer placement.
+
+The DC dialect has been heavily influenced by the Handshake dialect, and can either be seen as a successor to it, or a lower level
+abstraction. As of writing, DC is _fully deterministic_. This means that non-deterministic operators such as the ones found in Handshake - `handshake.merge, handshake.control_merge` - do **not** have a lowering to `dc`. `handshake` programs must therefore be converted or by construction not contain any of these non-deterministic operators. Apart from that, **all** handshake operations can be lowered to a combination of DC
```

Comment:
This is a nitpick, but you use "DC" and "Handshake" to talk about the dialects, which is fine by me, but then later you use "\`dc\`" and "\`handshake\`". I'd just stick to the former style.